util/check_tool_requirements.py

```diff
@@ -68,6 +68,34 @@ def get_verilator_version():
         return None
 
 
+def convert_verible_version(version_string):
+    '''
+    Convert Verible version string into a format compatible with
+    StrictVersion().
+    '''
+    # Drop the hash suffix and convert into version string that
+    # is compatible with StrictVersion in check_version below.
+    # Example: 0.0-808-g1e17daa -> 0.0.808
+    return '.'.join(version_string.split('-')[0:2])
+
+
+def get_verible_version():
+    '''Run Verible to check its version'''
+    try:
+        version_str = subprocess.run('verible-verilog-lint --version', shell=True,
+                                     check=True, stdout=subprocess.PIPE,
+                                     stderr=subprocess.STDOUT,
+                                     universal_newlines=True)
+        # Select the correct line in the output and drop the "v" prefix.
+        full_string = version_str.stdout.split(' ')[0].strip()[1:]
```

Comment:
If something comes unstuck and `version_str` ends up empty, we're going to get rather confusing errors down the road. Maybe it's best to pass the `v` prefix in to `convert_verible_version` (by removing the `[1:]` here) and then doing a quick check there with a regex, printing out a helpful error if we fail. Perhaps something like `v([0-9]+)\.([0-9]+)-([0-9]+)-g[0-9a-f]+$` (and you can use the captured groups instead of calling split).

Reply:
Good idea! I'll change that.